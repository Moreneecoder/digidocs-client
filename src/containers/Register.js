import { useState } from 'react';
import { registerUser } from '../hooks/http';

const Register = () => {
  const [isDoctor, setIsDoctorParams] = useState(false);
  const [name, setName] = useState('');
  const [phone, setPhone] = useState('');
  const [email, setEmail] = useState('');
  const [address, setAddress] = useState('');

  const handleCheckChange = (e) => {
    setIsDoctorParams(e.target.checked);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    const userData = {
      name, phone, email, address, is_doctor: isDoctor,
    };
    registerUser('http://digidocs-api.herokuapp.com/api/v1/users', userData);
    console.log(userData);
  };

  const handleInputChange = (data, from) => {
    switch (from) {
      case 'nameInput':
        setName(data.target.value);
        break;
      case 'phoneInput':
        setPhone(data.target.value);
        break;
      case 'emailInput':
        setEmail(data.target.value);
        break;
      case 'addressInput':
        setAddress(data.target.value);
        break;
      default:
        break;
    }
  };

  return (
    <div className="Register container">
      <form onSubmit={(e) => handleSubmit(e)}>
        <input className="form-control mt-1" required placeholder="Name" onChange={(e) => handleInputChange(e, 'nameInput')} />
        <input className="form-control mt-1" required placeholder="Phone" onChange={(e) => handleInputChange(e, 'phoneInput')} />
        <input className="form-control mt-1" required placeholder="Email" onChange={(e) => handleInputChange(e, 'emailInput')} />

        <div className="form-check">
          <span className="fw-bolder">Are You Regsitering As A Doctor?</span>
          <input className="form-check-input" type="checkbox" onChange={(e) => { handleCheckChange(e); }} id="is-doctor-check" />
        </div>

        {isDoctor ? <input className="form-control mt-1" required placeholder="Address" onChange={(e) => handleInputChange(e, 'addressInput')} /> : ''}

        <button type="submit" className="btn btn-success mt-1">Register</button>
      </form>
    </div>
  );
};

export default Register;
