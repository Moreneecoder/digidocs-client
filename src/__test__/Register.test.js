import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Register from '../containers/Register';

describe('Register Component', () => {
  const initialState = [];
  const mockStore = configureStore();
  let store;

  test('Renders Register component as expected', () => {
    store = mockStore(initialState);
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}><Register /></Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
