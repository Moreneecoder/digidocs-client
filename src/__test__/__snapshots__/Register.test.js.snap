// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component Renders Register component as expected 1`] = `
<div
  className="Register container pt-4"
>
  <form
    className="card mt-5 p-4 offset-md-2 col-md-8 offset-lg-3 col-lg-6"
    onSubmit={[Function]}
  >
    <div
      className="mb-3"
    >
      <span>
        UserName
      </span>
      <input
        aria-describedby="nameHelp"
        className="form-control"
        onChange={[Function]}
        placeholder="Name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="mb-3"
    >
      <span>
        Phone
      </span>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Phone"
        required={true}
        type="text"
      />
    </div>
    <div
      className="mb-3"
    >
      <span>
        Email address
      </span>
      <input
        aria-describedby="emailHelp"
        className="form-control"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
      />
      <div
        className="form-text main-text-color"
        id="emailHelp"
      >
        We'll never share your email with anyone else.
      </div>
    </div>
    <div
      className="mb-3 form-check"
    >
      <input
        className="form-check-input"
        id="is-doctor-check"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="fw-bolder"
      >
        Are You Regsitering As A Doctor?
      </span>
    </div>
    
    <button
      className="btn text-white main-bg-color"
      type="submit"
    >
      Register
    </button>
    <hr />
    <div
      className="text-end"
    >
      <span>
        Already have an account?
      </span>
       
      <a
        className="main-bg-color text-decoration-none text-white me-2 py-2 px-3"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </form>
</div>
`;
